src/python/py/models/builder.py

```diff
@@ -1495,6 +1506,125 @@ def make_mlp_fc(self, layer_id, mlp, root_input):
         # Assign output 0 of MLP layer as output of last layer
         self.mlp_attrs["output_0"] = f"{fc2_add_name}/output_0"
 
+    def make_block_sparse_moe(self, layer_id, bsm, root_input):
+        # Make nodes for the MoE subgraph
+        #
+        #                  root_input
+        #                 /       \
+        #         router_MatMul    |
+        #             /     \      |
+        #         Shape      |     |
+        #           |        |     |
+        #         Gather     |     |
+        #           |        |     |
+        #       Unsqueeze    |     |
+        #           |        |    /
+        #        Concat     /    /
+        #             \    /    /
+        #             Reshape  /
+        #                 \   /
+        #                  MoE
+        #                   |
+        #                 output
+        num_experts = self.moe_attrs["num_experts"]
+        top_k = self.moe_attrs["top_k"]
+        activation_type = self.moe_attrs["activation_type"]
+        normalize_routing_weights = self.moe_attrs["normalize_routing_weights"]
+        use_sparse_mixer = self.moe_attrs["use_sparse_mixer"]
+        use_int4 = self.moe_attrs["use_int4"]
+
+        gate_ops_base = f"/model/layers.{layer_id}/moe/gate/"
+        moe_name = f"/model/layers.{layer_id}/moe"
+
+        # Make MoE nodes
+        gate_name = gate_ops_base + "MatMul"
+        self.make_matmul(bsm.gate, gate_name, root_input)
+
+        shape_name = gate_ops_base + "Shape"
```

Comment:
Can we create all of the node names with f-string formatting (e.g. `f"{gate_ops_base}/Shape"`) to stay consistent with how other names are created?